src/operation/problem_results.jl

```diff
@@ -411,32 +454,135 @@ function read_realized_aux_variables(res::ProblemResults)
     return _read_realized_results(res.aux_variable_values, nothing)
 end
 
-function read_realized_variables(res::ProblemResults, variable_keys)
-    return _read_realized_results(
-        res.variable_values,
-        [VariableKey(x...) for x in variable_keys],
+"""
+Return the values for the requested variable keys for a problem.
+Accepts a vector of keys for the return of the values. If the time stamps and keys are
+loaded using the [load_results!](@ref) function it will read from memory.
+
+# Arguments
+- `variables::Vector{<:OptimizationContainerKey}` : keys of desired results
+- `initial_time::Dates.DateTime` : initial time of the requested results
+- `len::Int`: length of results
+"""
+function read_realized_variables(res::ProblemResults; kwargs...)
+    return read_realized_variables(res, collect(keys(res.variable_values)); kwargs...)
+end
+
+function read_realized_variables(res::ProblemResults, variables; kwargs...)
+    return read_realized_variables(res, [VariableKey(x...) for x in variables]; kwargs...)
+end
+
+function read_realized_variables(
+    res::ProblemResults,
+    variables::Vector{<:OptimizationContainerKey};
+    initial_time::Union{Nothing, Dates.DateTime} = nothing,
+    count::Union{Int, Nothing} = nothing,
+)
+    result_values = read_realized_variables_with_keys(
+        res,
+        variables;
+        initial_time = initial_time,
+        count = count,
     )
+    return Dict(encode_key_as_string(k) => v for (k, v) in result_values)
 end
 
-function read_realized_variables(res::ProblemResults)
-    return _read_realized_results(res.variable_values, nothing)
+function read_realized_variables_with_keys(
```

Comment:
Is this function private to this file? If so, `_read_realized_variables_with_keys`

Reply:
It's used in PowerGraphics. It definately doesn't need to be exported, but should I still prefix with "_"?

Reply:
Then I think it is good as-is.